### PR TITLE
Gl: Use common code for fragment constants.

### PR DESCRIPTION
**Pull request by vlj**

### CODE REVIEW

**Comment by O1L** [comment]
Nice!
I found another minor bug here (implicit cast): https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/Common/VertexProgramDecompiler.cpp#L332 and https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/Common/VertexProgramDecompiler.cpp#L337
It fixes black screens in some games. And set viewport here https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/GL/GLGSRender.cpp#L589 fixes the area of visibility.

File: rpcs3/Emu/RSX/GL/GLFragmentProgram.cpp [hunk 1]
```diff
@@ -68,13 +68,18 @@ void GLFragmentDecompilerThread::insertConstants(std::stringstream & OS)
 
 	}
 
+	OS << "layout(std140, binding = 2) uniform FragmentConstantsBuffer" << std::endl;
+	OS << "{" << std::endl;
 	for (ParamType PT : m_parr.params[PF_PARAM_UNIFORM])
```
Comment (paulsapps) [inline rpcs3/Emu/RSX/GL/GLFragmentProgram.cpp]: Why not reference?

Comment (vlj) [inline rpcs3/Emu/RSX/GL/GLFragmentProgram.cpp]: what do you mean ??

Comment (paulsapps) [inline rpcs3/Emu/RSX/GL/GLFragmentProgram.cpp]: I dunno what happen there, I added the comment on a line that had something like (for PT : list) {}

Comment (danilaml) [inline rpcs3/Emu/RSX/GL/GLFragmentProgram.cpp]: @vlj why not `const ParamType& PT` (here and other places). (and same for ParamItem, copying is unnecessary)

**Comment by vlj** [comment]
@O1L which implicit cast ? adding std::string and const char is not correctly handled ?

**Comment by vlj** [comment]
Fixed viewport and ref in loops

**Comment by O1L** [comment]
@vlj 
getIntTypeName(4) - getFloatTypeName(4) -> implicit cast and shaders compilation failed. Must be
return m_parr.AddParam(PF_PARAM_NONE, getIntTypeName(4), "a" + std::to_string(d0.addr_reg_sel_1), getIntTypeName(4) + "(0, 0, 0, 0)") + AddAddrMask();

**Pull request merged**